include/wasm/WasmModule.h

```diff
@@ -129,6 +133,9 @@ class WasmModule
     int stdoutMemFd;
     ssize_t stdoutSize;
 
+    std::mutex moduleMemoryMutex;
+    std::mutex moduleStateMutex;
```

Comment:
These two mutexes are required as things can access module memory directly, but state stuff may also need to access memory too (while already holding the state lock). I could have used a single module-level recursive lock, but I think this is clearer and easier to debug when things go wrong.